pkg/piecestore/pstore.go

```diff
@@ -20,86 +19,70 @@ var (
 	FSError  = errs.Class("fsError")
 )
 
-// PathByHash -- creates datapath from hash and dir
-func PathByHash(hash, dir string) (string, error) {
-	if len(hash) < 20 {
-		return "", ArgError.New("Invalid hash length")
+// PathByID -- creates datapath from id and dir
+func PathByID(id, dir string) (string, error) {
+	if len(id) < 20 {
+		return "", ArgError.New("Invalid id length")
+	}
+	if dir == "" {
+		return "", ArgError.New("No path provided")
 	}
 
-	folder1 := string(hash[0:2])
-	folder2 := string(hash[2:4])
-	fileName := string(hash[4:])
+	folder1 := string(id[0:2])
+	folder2 := string(id[2:4])
+	fileName := string(id[4:])
 
 	return path.Join(dir, folder1, folder2, fileName), nil
 }
 
-// Store -- Store data into piece store
-// 	hash 					(string)		Hash of the data to be stored
-// 	r 			  		(io.Reader)	File/Stream that contains the contents of the data to be stored
-// 	length 				(length)		Size of the data to be stored
-// 	psFileOffset 	(offset)  	Offset of the data that you are writing. Useful for multiple connections to split the data transfer
-// 	dir 					(string)		pstore directory containing all other data stored
-// 	returns 			(error) 		error if failed and nil if successful
-func Store(hash string, r io.Reader, length int64, psFileOffset int64, dir string) (int64, error) {
+// StoreWriter -- Store data into piece store in multiple writes
```

Comment:
Comments should be sentences. The nice thing about typed languages is you don't need to specify the type in comments.

Reply:
done